File: yh_sum/src/yh_sum_sub.cpp
#include "ros/ros.h" 
#include "yh_sum/yh_sum_msg.h"  

// class MyClass
// {
//     public:
//         MyClass()
//         {
//             sub = nh.subscribe("yh_sum_topic", 100, &MyClass::msgCallback, this);
//         }
//         void msgCallback(const yh_sum::yh_sum_msg::ConstPtr &msg)
//         {
//             printf("%d\n", old_data + msg->data);
//             old_data = msg->data;
//         }

//     private:
//         ros::NodeHandle nh;
//         ros::Subscriber sub;
//         int old_data = 0;
// };


int old_data = 0;

void msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg)   
{
    printf("%d", old_data + msg->data);
    old_data = msg->data;
}


int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_sum_sub"); 
    ros::NodeHandle nh; 

    ros::Subscriber sub = nh.subscribe("yh_sum_topic", 100, msgCallback);   
                                                                          
    // MyClass my_class;

    ros::spin();   
                    

    return 0;

}

File: yh_sum/scripts/yh_sum_sub.py
#!/usr/bin/python3
# -*- coding: utf8 -*-

import imp
import rospy
from yh_sum.msg import yh_sum_msg


class SumClass:
    def __init__(self) -> None:
        self.sub = rospy.Subscriber("yh_sum_topic", yh_sum_msg, self.msgCallback)
        self.old_data = 0

    def msgCallback(self, msg):
        sum = self.old_data + msg.data
        self.old_data = msg.data
        print(sum)



# sum = 0
# # old_data = 0

# def msgCallback(msg):

#     rospy.loginfo("number is = %d", msg.data)

#     num = msg.data
    
#     global sum

#     sum += num

#     print(sum)

    """
    global old_data
    print(old_data + msg.data)
    old_data = msg.data
    """


def listener():
    rospy.init_node("yh_sum_sub", anonymous=True)
    # rospy.Subscriber("yh_sum_topic", yh_sum_msg, msgCallback)
    sum_class = SumClass()
    rospy.spin()

if __name__ == "__main__":
    listener()
